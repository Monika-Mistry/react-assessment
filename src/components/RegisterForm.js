import React from 'react';
import {
    Form,
    FormGroup,
    Label,
    Input,
    Col,
    Button,
    FormFeedback
} from 'reactstrap';


export default function RegisterForm(props) {
    const username = props.errors.username;
    const email = props.errors.email;
    const password = props.errors.password;
    const password2 = props.errors.password2;
    const isAdded = props.isAdded;

    return (
        <div>
            <Form className="container" onSubmit={props.addUser}>
                <FormGroup row>
                    <Label for="username" sm={2}>Username:</Label>
                    <Col sm={10}>
                        <Input
                            required
                            type="text"
                            name="username"
                            id="username"
                            placeholder="Enter your username"
                            invalid={username !== undefined}
                            valid={isAdded}
                        />
                        <FormFeedback invalid>{username}</FormFeedback>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label for="email" sm={2}>Email:</Label>
                    <Col sm={10}>
                        <Input
                            required
                            type="email"
                            name="email"
                            id="email"
                            placeholder="Enter your email"
                            invalid={email !== undefined}
                            valid={isAdded}
                        />
                        <FormFeedback invalid>{email}</FormFeedback>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label for="password" sm={2}>Password:</Label>
                    <Col sm={10}>
                        <Input
                            required
                            type="password"
                            name="password"
                            id="password"
                            placeholder="Enter your password"
                            invalid={password !== undefined}
                            valid={isAdded}
                        />
                        <FormFeedback invalid>{password}</FormFeedback>
                    </Col>
                </FormGroup>
                <FormGroup row>
                    <Label for="password2" sm={2}>Re-enter Password:</Label>
                    <Col sm={10}>
                        <Input
                            required
                            type="password"
                            name="password2"
                            id="password2"
                            placeholder="Enter your password again"
                            invalid={password2 !== undefined}
                            valid={isAdded}
                        />
                        <FormFeedback invalid>{password2}</FormFeedback>
                    </Col>
                </FormGroup>
                <div className="text-center">
                    <Button>Register</Button>
                </div>


            </Form>
            <br />
            {isAdded && <h1 className="text-center text-success">User Added</h1>}
        </div>
    )
}