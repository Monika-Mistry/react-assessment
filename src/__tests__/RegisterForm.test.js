import React from 'react';
import renderer from 'react-test-renderer';

import RegisterForm from '../components/RegisterForm';



it('renders a register form', () => {
    const wrapper = renderer.create(<RegisterForm />).toJSON();
    expect(wrapper).toMatchSnapshot();

})

it('renders a successful registration', () => {
    const wrapper = renderer.create(<RegisterForm isAdded={true} />).toJSON();
    expect(wrapper).toMatchSnapshot();
})