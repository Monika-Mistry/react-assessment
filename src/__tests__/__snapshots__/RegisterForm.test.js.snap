// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a register form 1`] = `
<div>
  <form
    className="container"
  >
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="username"
      >
        Username:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="username"
          name="username"
          placeholder="Enter your username"
          required={true}
          type="text"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="email"
      >
        Email:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="email"
          name="email"
          placeholder="Enter your email"
          required={true}
          type="email"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="password"
      >
        Password:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="password"
          name="password"
          placeholder="Enter your password"
          required={true}
          type="password"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="password2"
      >
        Re-enter Password:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="password2"
          name="password2"
          placeholder="Enter your password again"
          required={true}
          type="password"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Register
      </button>
    </div>
  </form>
  <br />
  <h1
    className="text-center text-success"
  >
    User Added
  </h1>
</div>
`;

exports[`renders a successful registration 1`] = `
<div>
  <form
    className="container"
  >
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="username"
      >
        Username:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="username"
          name="username"
          placeholder="Enter your username"
          required={true}
          type="text"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="email"
      >
        Email:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="email"
          name="email"
          placeholder="Enter your email"
          required={true}
          type="email"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="password"
      >
        Password:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="password"
          name="password"
          placeholder="Enter your password"
          required={true}
          type="password"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="row form-group"
    >
      <label
        className="col-sm-2 col-form-label"
        htmlFor="password2"
      >
        Re-enter Password:
      </label>
      <div
        className="col-sm-10"
      >
        <input
          className="is-valid form-control"
          id="password2"
          name="password2"
          placeholder="Enter your password again"
          required={true}
          type="password"
        />
        <div
          className="invalid-feedback"
          invalid={true}
        />
      </div>
    </div>
    <div
      className="text-center"
    >
      <button
        aria-label={null}
        className="btn btn-secondary"
        onClick={[Function]}
      >
        Register
      </button>
    </div>
  </form>
  <br />
  <h1
    className="text-center text-success"
  >
    User Added
  </h1>
</div>
`;
